testui/elements/testui_element.py

```diff
@@ -205,41 +210,40 @@ def visible_for(self, seconds=1):
                            f'for {seconds}s')
         return self
 
-    def wait_until_visible(self, seconds=10.0, log=True):
+    def wait_until_visible(self, seconds=10.0, log=True) -> "Elements":
         start = time.time()
-        err = None
-        err_text = 'not'
+
+        is_not = self.__is_not
+
+        # Executes the loop at least once in case `seconds` <= 0
+        entered_loop = False
+        while time.time() < start + seconds or not entered_loop:
+            if self.is_visible(log=False):
+                if log:
+                    self.__put_log(
+                        f'{self.device_name}: Element "{self.locator_type}: {self.locator}" '
+                        f"was visible in {time.time() - start}s"
+                    )
+
+                return self
+
+            self.__is_not = is_not
```

Comment:
Because `is_visible()` sets `self.__is_not = False` every time it is called. For this reason, to not loose the original value when `wait_until_visible()` is called, it is reset.

Reply:
maybe it would be better to introduce `**kwargs` to is_visible defining one kwarg - "is_not" that would allow to overwrite self.__is_not

Reply:
I am not a fan of `**kwargs` because they are useful for an unknown number of arguments (which is not the case for this function) + available arguments have to be explicitly and clearly documented. 

A better approach would be to use optional parameters. 

But the best approach I think would be to rethink `self.__is_not` logic completely (deserves a separate issue) because handling it is a mess + your suggested approach would do basically the same as what is being done now but done in `is_visible()` scope.

Reply:
well, obviously the best would be to refactor the `is_not` logic, but i don't have any ideas on that. the reason I think that kwargs could be a solution here is exactly the ability to provide an unnamed argument - external users shouldn't be using it so i wouldn't want to create a full function argument for something that is used once

Reply:
Okay, updated the parameters for `is_visible()`. This might be a good approach **for now**. Good thing about `**kwargs` is that they can be easily removed from the function parameters without any breaking changes